### PR TITLE
new image processing function: convert RGB to greyscale

### PR DESCRIPTION
**Pull request by hiovidiu**
Conversion function from a float RGB array to its greyscale representation.
R=G=B=0.299f*R+0.587f*G+0.114f*B used. Uses fmaf().
Added test suite 'image' for testing rgb2greyscale. Inspired from 'dsp' test suite.

### CODE REVIEW

File: src/image/p_grayscale.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include <pal.h>
+
+/**
+ * RGB to Grayscale converter
+ *
+ * Each channel is replaced with the grayscale value.
+ * The grayscale value is computed as 0.299*R + 0.587*G + 0.114*B
```
Comment (tcptomato) [inline src/image/p_grayscale.c]: Might be an idea to give the coefficients as an argument to the function. The current static coeficients are for BT.601, HDTV ( BT. 709) has them slightly different.

**Comment by lchamon** [comment]
With coefficients as arguments this function would become a special case of `fir_decim` (`x` is the image, `h` are the coefficients, `nh = 3`, and `dec` = 3). `p_grayscale` could use it internally (I'm still not done with it though ;) ). Still, the coefficients could be hardcoded somehow to avoid having to look them up everytime.

Small question: I've seen `__builtin_fmaf` used a few times around. Is that from the epiphany core? As far as portability goes, should that be used?

**Comment by aolofsson** [comment]
Yes, the payoff versus general 2D filter is pretty big, so it's worth it. Nobody would do grayscale conversion as a filter. I think the fmaf is from gcc, but Epiphany, PowerPC, and many others have a built in fused multiply add instruction.  Should work on architectures without fmaf as well but haven't tested it.

**Comment by lchamon** [comment]
Oh no... Sure, no one should ever use a filter for that! I was just suggesting that the internal implementation could use filtering. But only if the coefficients weren't hard coded.

Well living and learning! I tested out `fmaf` in the FIR functions, but it makes things 4-5x slower, although it's slightly more precise.

**Pull request merged**